classes/04class/exercises/c04-iac04/mrcsmonteiro/terraform-code/iac-04-module/alb/main.tf

```diff
@@ -0,0 +1,32 @@
+resource "aws_alb" "alb" {
```

Comment:
Hi @mrcsmonteiro , I got the pattern you are following here and it is more than the exercise asked, that's fine. The exercise is about creating 1 module that has ALB, ASG, alarms, etc.
Probably you noticed the overhead you had for breaking it down every one (or two) resource types per module.

From TF docs:
> We do not recommend writing modules that are just thin wrappers around single other resource types. If you have trouble finding a name for your module that isn't the same as the main resource type inside it, that may be a sign that your module is not creating any new abstraction and so the module is adding unnecessary complexity. Just use the resource type directly in the calling module instead.

https://www.terraform.io/docs/modules/index.html#when-to-write-a-module